File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        ans = 0 
        count = [0]*26
        left = 0 
        for i in range(len(s)):
            count[ord(s[i])-ord('A')]+=1
            sum = 10000
            index = 0
            while(left<i and k<sum-count[index]):
                index = 0 
                sum = 0
                for j in range(26):
                    sum+=count[j]
                    if (count[j]>count[index]):
                        index = j
                if (k>=sum-count[index]):
                    break
                else :
                    count[ord(s[left])-ord('A')]-=1
                    left+=1
            ans = max(i - left+1,ans)
        return ans                        

File: Data Structures & Algorithms/max-area-of-island/submission-0.cpp
class Solution {
public:
int df[5] = {0,1,0,-1,0};

    int dfs(vector<vector<int>> &grid, int i , int j){
        if (grid[i][j]==0){
           return 0;
        }
        grid[i][j]=0;
        int sum = 0 ; 
        for (int k=0;k<4;++k){
            int l = i + df[k];
            int r = j + df[k+1];
            if (l<0 || r <0 || l>=grid.size() || r>=grid[0].size()){
                continue;
            }
            sum+=dfs(grid,l,r);
        }
        return sum+1;
    }

    int maxAreaOfIsland(vector<vector<int>>& grid) {
                int count = 0 ; 
        for (int i=0;i<grid.size();++i){
            for (int j=0;j<grid[0].size();++j){
                if (grid[i][j]==1){
                    count = max(count,dfs(grid,i,j));
                }
            }
        }
        return count;
    }
};
